flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/stream/sql/DeduplicateTest.xml

```diff
@@ -207,7 +207,7 @@ LogicalProject(a=[$0], b=[$1], c=[$2], proctime=[$3], rowtime=[$4], rowNum=[$5])
     </Resource>
     <Resource name="optimized exec plan">
       <![CDATA[
-Calc(select=[a, b, c, PROCTIME_MATERIALIZE(proctime) AS proctime, rowtime, 1:BIGINT AS rowNum])
+Calc(select=[a, b, c, PROCTIME_MATERIALIZE(proctime) AS proctime, rowtime, 1:BIGINT AS $5])
```

Comment:
please create a JIRA to trace the change ?

Reply:
In the previous materialize version, the last field of the new added project is rowNum
![image](https://user-images.githubusercontent.com/1525333/128148720-30eec7cb-3080-45bf-803c-a9d24dcb4505.png)

In the new version, materialize move to the later phase. after `project_rewrite` phase, the last field of LogicalProject rowtype is $5 instead of rowNum.
So  the last field of the new added calc is $5
![image](https://user-images.githubusercontent.com/1525333/128148392-59d0f520-db2e-40d7-8ee7-e8b6038eacb9.png)

Reply:
After some investigation, I found the root cause is original relnode type field names are not equals to that of equivalent relnode after `ProjectToWindowRule.PROJECT_TO_LOGICAL_PROJECT_AND_WINDOW`. It seems not to be reasonable, however calcite does not have a strong guarantee that field type names are same after apply rule .

Reply:
I create a similar issue in Calcite community: https://issues.apache.org/jira/browse/CALCITE-2718. Julian's point of view, "We don't guarantee to preserve names when planner rules are applied." But I think, we should to improve it as much as possible to make the plan easier to read, because the names will be display in execution graph on flink web.

Reply:
I agree with you. However it's hard if Calcite does not provide strong guarantee. Because we could only guarantee the extended rules in Flink obey this,  but currently we also introduces many built-in rules of Calcite in Flink.

Reply:
I create a JIRA [FLINK-23638](https://issues.apache.org/jira/browse/FLINK-23638) to track the issue.